src/main/kotlin/Main.kt

```diff
@@ -1,15 +1,55 @@
-fun main(args: Array<String>) {
-    println("Hello world!")
+import java.util.ArrayList
+
+/*
+* Implementation disjoint set union
+* */
+
+// global data for DSU
+var parent = ArrayList<Int>()
+var rank = ArrayList<Int>()
+var countElement = 0
```

Comment:
global data is a bad idea

Reply:
I thought I can't use class, because it's new lesson. 
Ok, I'll warp DSU-algorithm in object.